packages/core/src/controllers/SwapController.ts

```diff
@@ -655,16 +660,20 @@ export const SwapController = {
 
   // -- Send Transactions --------------------------------- //
   async sendTransactionForApproval(data: TransactionParams) {
-    const { fromAddress } = this.getParams()
+    const { fromAddress, isAuthConnector } = this.getParams()
 
     state.loadingApprovalTransaction = true
-    RouterController.pushTransactionStack({
-      view: null,
-      goBack: true,
-      onSuccess() {
-        SnackController.showLoading('Approving transaction...')
-      }
-    })
+    if (isAuthConnector) {
+      RouterController.pushTransactionStack({
+        view: null,
+        goBack: true,
+        onSuccess() {
+          SnackController.showLoading('Approving transaction...')
+        }
+      })
+    } else {
+      SnackController.showLoading('Approve spending increase')
```

Comment:
Do we need any loading indicator on the screen as well?

Reply:
Yes, this is just a feedback to the users that let them know they need to approve the TX, toast will be visible for few seconds.